src/sciline/task_graph.py

```diff
@@ -76,6 +74,11 @@ def visualize(
         kwargs:
             Keyword arguments passed to :py:class:`graphviz.Digraph`.
         """
+        from IPython.display import display_html
+
         from .visualize import to_graphviz
 
-        return to_graphviz(self._graph, **kwargs)
+        display_html(
+            to_graphviz(self._graph, **kwargs).pipe(format='svg', encoding='utf8'),
+            raw=True,
+        )
```

Comment:
I thought the visualization function should only be used for visualizing the graph, and if IPython is not avaliable we cannot do that, so it seemed sensible to me to make that explicit.

If the user want to access the `graphviz` graph should they not use the `to_graphviz` function instead?

Reply:
> and if IPython is not avaliable we cannot do that, so it seemed sensible to me to make that requirement explicit.

Why is IPython required?

Reply:
It's IPython that implements the logic that takes our html representation and renders it to the user.

Reply:
Yes, but Graphviz can render directly to a file, no need for IPython. There are two steps here:

1. Make a Graphviz graph.
2. Make an HTML representation.

The second step happens automatically if you output the graph (by writing it at the end of a cell). But you can just call `task_graph.visualize().render(...)` from a Python script and never reach step two.

Reply:
It makes a lot of sense to enable users to create `graphviz` objects from their task graphs.
The issue is that the HTML representation created by the `graphviz` object is not good because we can't copy paste from it. I've looked into it and it seems that cannot be changed by changing some setting on the `graphviz` object.

Do you think it might be a good idea to add a `_repr_html_` method directly to the `TaskGraph` class?

Then the users would not even have to call the `visualize` method to view their graphs.
The problem then is that we can't pass `graphviz` kwargs before drawing the graph.

We could also add a separate `display` method that takes graphviz kwargs just as visualize does now. But having both `visualize` and `display` seems confusing.

Reply:
> Do you think it might be a good idea to add a `_repr_html_` method directly to the `TaskGraph` class?
> 
> Then the users would not even have to call the `visualize` method to view their graphs. The problem then is that we can't pass `graphviz` kwargs before drawing the graph.

I think adding such a `_repr_html_` would be useful, but it does unfortunately not solve entire the problem: We also have `Pipeline.visualize(MyType)`, which is basically a shortcut for `Pipeline.get(MyType).visualize()`. Adding `_repr_html_` also to `Pipeline` is not really feasible, since we do not know which type(s) to generate the graph for.
 
> We could also add a separate `display` method that takes graphviz kwargs just as visualize does now. But having both `visualize` and `display` seems confusing.

Note that I used `visualize()` in Sciline because this is the API Dask uses, i.e., some users may be familiar: https://docs.dask.org/en/stable/graphviz.html.

Reply:
> I think adding such a _repr_html_ would be useful, but it does unfortunately not solve entire the problem: We also have Pipeline.visualize(MyType), which is basically a shortcut for Pipeline.get(MyType).visualize(). 

But why not just remove `Pipeline.visualize` and let people use `Pipeline.get` instead? If we add the `_repr_html_` to `TaskGraph` we don't need to call `.visualize` on the returned `TaskGraph` anyway.

Reply:
> > I think adding such a _repr_html_ would be useful, but it does unfortunately not solve entire the problem: We also have Pipeline.visualize(MyType), which is basically a shortcut for Pipeline.get(MyType).visualize().
> 
> But why not just remove `Pipeline.visualize` and let people use `Pipeline.get` instead? If we add the `_repr_html_` to `TaskGraph` we don't need to call `.visualize` on the returned `TaskGraph` anyway.

That sounds like an option....  but on the other hand, maybe drawing a dot graph by default is not what we want? I assume graphs could get quite large, so maybe we want to use `TaskGraph._repr_html_` to provide an HTML summary, similar to how a Dask collection shows something like this:
<img width="514" alt="image" src="https://github.com/scipp/sciline/assets/12912489/525faa41-75a3-40b4-b6f7-40a2900e0dc6">
... instead of visualizing a potentially huge graph by default. We might want to use `TaskGroup._repr_html_` to, e.g., list the parameters that will be used, which providers, ... in a table-like manner (similar to the simple views you get from Scipp data structures).

Reply:
Example from Dask: 
<img width="599" alt="image" src="https://github.com/scipp/sciline/assets/12912489/18b8114c-9f64-4ca2-ac72-909d0ef46e7b">